### ISSUE TITLE
Deployment: Serverless option

### ISSUE THREAD

**Issue by I-Dont-Remember** [happybara-io/WorkflowBuddy#29]
Serverless is a common utility in the current age, and allowing people to deploy bots on FaaS providers like AWS Lambda is pretty much a given - whether that's because their company doesn't allow use of external hosting like Fly, or they want to manage costs, or etc.

AWS Lambda is likely the largest contingent, so this work would include:
- modifying core code to operate in the constraints of an ephemeral function (really just updating Bolt references, or potentially removing entirely and going with the bare SDK). If we abstract enough of the code into utility functions, they could be shared across multiple deployment types.
- Setting up a `serverless.yml` file to have any necessary infrastructure as code.

**Comment by jphorn** [comment]
Would this work as a Cloudflare Worker as well?

**Comment by I-Dont-Remember** [comment]
Yes, solid idea, especially with it having KV stores, D1 (sql database), the queues coming out, etc!

Would require a decent bit of work to flesh out though. I was [playing with the idea recently](https://github.com/I-Dont-Remember/cloudflare-workers-slack-bot) to see what it would take to have a fully-featured Slack App on Workers, ran into a few things:
- Bolt-js & even the `@slack/web-api` won't work out of the box since they are written for Node - things blew up when i tried using.
- Small team wrote a [convenient alternative](https://github.com/sagi/workers-slack) to `web-api` for use in Workers and open-sourced it - works great, but may need contributions to unwrap all of the Slack functionality.

Though it sounds like a lot of work to do a rewrite, it actually wouldn't be **TOO** crazy in this case. A lot of the logic is very simple, the blocks code is language-agnostic, so is definitely possible.
